modules/shared/src/main/scala/org/tessellation/security/signature/SignedValidator.scala

```diff
@@ -79,6 +84,17 @@ object SignedValidator {
       else
         NotEnoughSignatures(signed.proofs.size, minSignatureCount).invalidNec
 
+    def isSignedBy[A: Encoder](
+      signed: Signed[A],
+      signerAddress: Address
+    ): F[SignedValidationErrorOr[Signed[A]]] =
+      signed.proofs.existsM { proof =>
+        proof.id.toAddress.map(_ === signerAddress)
+      }.ifM(
+        signed.validNec[SignedValidationError].pure[F],
+        Validated.invalidNec[SignedValidationError, Signed[A]](NotSignedByAddressOwner).pure[F]
+      )
```

Comment:
If we use `.ifF` instead then we don't need to lift the results with `pure`.
Or we can simplify it a bit further by doing this?
```suggestion
      signed.proofs
        .existsM(_.id.toAddress.map(_ === signerAddress))
        .map(Validated.condNec(_, signed, NotSignedByAddressOwner))
```

Reply:
Yeah the less code the better. It makes sense to use `ifF` here.

Reply:
Yeah it's simpler I will use that, thanks.